### DISCUSSION TITLE
Does jotai have a limitation caused by its WeakMap usage when it comes to multiple window apps?

### DISCUSSION THREAD

**Discussion by unbeauvoyage** [pmndrs/jotai#2748] [Q&A] [answered]
I had migrated to jotai and been building my app on it until I came across, [in a discussion](https://www.reddit.com/r/reactjs/comments/185z3dl/why_should_you_use_zedux_over_jotai/), the below "limitation" of jotai when it comes to its usage of WeakMap to hold references (when making a multiple window app, which is what I am making). Is this a valid concern, or a misunderstanding on the part of the person who wrote it? If so, is it actually easy to overcome that limitation?

> Jotai was not an option for our fintech apps because of its WeakMap approach. Fintech apps make use of multiple desktop windows a lot. WeakMaps rely on object reference sharing to find cached values. Sharing every atom config object's reference across all child windows is doable, but it's pretty impractical.

**Comment by AjaxSolutions** [comment]
Can you use atomWithBroadcast?

**Reply by unbeauvoyage** [reply]
Thanks, I'll look into this.

**Comment by dai-shi** [comment]
I don't think it's because of WeakMap but the store object. If there are two memory spaces, two stores are isolated. Like, `globalThis.foo = 1` in one window and `globalThis.foo = 2` in another window. Am I getting the problem correctly?

**Reply by unbeauvoyage** [reply]
I guess the original quote I brought to your attention might be implying that since jotai's atoms' identities are tied to object references in memory because of WeakMap (despite bringing other benefits in terms of memory leaks etc., as you have pointed out on other occasions), they are not as easily serializable as, say, zustand's approach. The quote also talks about it being doable, though impractical. However, I wasn't sure about the implications, hence my question.

**Reply by dai-shi** [reply, marked as answer]
> they are not as easily serializable

That's correct. It's doable but not easy.
